clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +33,72 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	defer cleanupResources(ctx)
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		ste.ExecuteStep(ctx, &conf.Steps[i])
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		defer wg.Wait()
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				ste.ExecutePhase(ctx, phase)
+			})
+		}
+	}
 	return nil
 }
 
 // ExecutePhase executes single test phase based on provided phase configuration.
 func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) error {
+	if phase.NamespaceRange == nil {
+		for j := range phase.ObjectBundle {
+			innerExecuteObject(ctx, &phase.ObjectBundle[j], phase.ReplicasPerNamespace, "")
+		}
+	} else {
+		nsBasename := framework.AutomanagedNamespaceName
+		if phase.NamespaceRange.Basename != nil {
+			nsBasename = *phase.NamespaceRange.Basename
+		}
+		for i := phase.NamespaceRange.Min; i <= phase.NamespaceRange.Max; i++ {
+			nsName := fmt.Sprintf("%v-%d", nsBasename, i)
+			for j := range phase.ObjectBundle {
+				innerExecuteObject(ctx, &phase.ObjectBundle[j], phase.ReplicasPerNamespace, nsName)
+			}
+		}
+	}
 	return nil
 }
 
 // ExecuteObject executes single test object operation based on provided object configuration.
 func (ste *simpleTestExecutor) ExecuteObject(ctx Context, object *api.Object) error {
```

Comment:
Could you formulate (in normal text) what is it that you expect `ExecuteObject` to do? It'll help you implement this.

Reply:
TBH, I want to implemented this part in separate PR and focus only on test, step and phase execution.

Reply:
I've merge `ExecuteObject` and `innerExecuteObject` into one function. Interface has been changed.